src/store/TimeperiodStore.js

```diff
@@ -14,11 +15,56 @@ export default class TimeperiodStore {
 
     @observable timeUnit = null;
     @observable interval = null;
+    @observable remain = "00:00";
 
     onContextReady = () => {
         const { timeUnit, interval } = this.context.stx.layout;
         this.timeUnit = getTimeUnit({ timeUnit, interval });
         this.interval = interval;
+
+        const stx = this.context.stx;
+
+        this.showCandleCountdown();
+
+        reaction(() => this.timeUnit , () => { this.showCandleCountdown(); });
+
+        const displayMilliseconds = (ms) => {
+            const totalSec = ms / 1000;
+            if (totalSec <= 0) { return null; }
+            const padNum = (n) => (`0${n}`).slice(-2);
+            const seconds = padNum(Math.trunc((totalSec) % 60));
+            const minutes = padNum(Math.trunc((totalSec/60) % 60));
+            let hours = Math.trunc((totalSec / 3600) % 24);
+            hours = hours ? `${hours}:` : '';
+            return `${hours}${minutes}:${seconds}`;
+        };
+
+        setInterval(() => {
+            var dataSet = stx.chart.dataSet;
+            if(dataSet && dataSet.length != 0 ){
+                let diff = new Date() - dataSet[dataSet.length-1].DT;
+                this.remain = displayMilliseconds((getIntervalInSeconds(stx.layout) * 1000) - diff);
+                this.remain = this.remain ? this.remain : "00:00";
+            }
+        }, 1000);
+    }
+
+    showCandleCountdown = () => {
```

Comment:
notice that there is a delay before the countdown shows up. This can be fixed by adding `stx.draw()` at the end of this method.

Reply:
@brucebinary what method? you mean setInterval?

Reply:
Done!